modin/core/dataframe/base/dataframe/dataframe.py

```diff
@@ -0,0 +1,506 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module contains class ModinDataframe.
+
+ModinDataframe is a parent abstract class for any dataframe class.
+"""
+
+from abc import ABC, abstractmethod
+from typing import List, Hashable, Optional, Callable, Union, Dict
+
+
+class ModinDataframe(ABC):
+    """
+    An abstract class that represents the Parent class for any DataFrame class.
+
+    This class is intended to specify the behaviours that a DataFrame must implement.
+
+    For more details about how these methods were chosen, please refer to this
+    (http://www.vldb.org/pvldb/vol13/p2033-petersohn.pdf) paper, which specifies
+    a DataFrame algebra that this class exposes.
+    """
+
+    @abstractmethod
+    def mask(
+        self,
+        row_labels: Optional[List[Hashable]] = None,
+        row_positions: Optional[List[int]] = None,
+        col_labels: Optional[List[Hashable]] = None,
+        col_positions: Optional[List[int]] = None,
+    ) -> "ModinDataframe":
+        """
+        Mask rows and columns in the dataframe.
+
+        Allow users to perform selection and projection on the row and column number (positional notation),
+        in addition to the row and column labels (named notation)
+
+        Parameters
+        ----------
+        row_labels : list of hashable, default: None
+            The row labels to extract.
+        row_positions : list of int, default: None
+            The row indices to extract.
```

Comment:
there's undefined behavior if both are specified, right? let's define it then: point which labels are prioritized (labels or positional based) or that it leads to an error.

Reply:
In the `Notes` section of this docstring, the following is specified:
```python
        If both `row_labels` and `row_positions` are set, `row_labels` will be used.
        The same rule applied to `col_labels` and `col_positions`.
```
should I delete this, and instead add checks that throw Exceptions if both are specified?